### DISCUSSION TITLE
Extract Headings only?

### DISCUSSION THREAD

**Discussion by MaPeras** [Zettlr/Zettlr#3975] [Q&A] [answered]
Hi folks,

looking for a way to extract only headings from a markdown file that consists of headings+text, in order to insert them into another markdown file.

background: I use Zettlr to write a manuscript for a seminar, but I also use it to create slides - usually with the headings from my manuscript (displayed with HackMD). As I do have many headings (lawyer!), copy/pasting them all into various slides is quite tedious.

Thanks a lot!
Max

**Comment by nathanlesage** [comment, marked as answer]
If you only want to extract the headings themselves, it should be easy to do with a Shell script that looks for lines starting with `###`. If you also want to extract whole sections, it's a little bit more involved. Feel free to check out Zettlr's approach to move sections: 

https://github.com/Zettlr/Zettlr/blob/7940d371d9643a77ea731e2eafedf2eaa3f74461/source/common/modules/markdown-editor/index.ts#L488-L520

And the ToC field:

https://github.com/Zettlr/Zettlr/blob/develop/source/common/modules/markdown-editor/plugins/toc-field.ts
